torchao/prototype/quantization/safetensors_JSONEncoders.py

```diff
@@ -0,0 +1,175 @@
+import dataclasses
+import json
+from typing import Any, Dict
+import enum
+
+import torch
+from torchao.quantization import Float8Tensor
+import importlib
+from torchao.quantization.granularity import PerRow
+from torchao.quantization.quantize_.workflows.float8.float8_tensor import (
+    QuantizeTensorToFloat8Kwargs,
+)
+
+ALLOWED_QUANT_DTYPES = {
+    "torch.float8_e4m3fn": torch.float8_e4m3fn,
+    # add to me
+}
+ALLOWED_GRANUALARITY = {"PerRow()": PerRow()}
+
+ALLOWED_AO_MODULES = {
+    "torchao.quantization",
+    "torchao.sparsity.sparse_api",
+    "torchao.prototype.quantization",
+    "torchao.prototype.mx_formats",
+    "torchao.dtypes",
+    "torchao.prototype.awq",
+    "torchao.quantization.quantize_.common",
+    "torchao.quantization.quantize_.workflows",
+}
+
+
+class Float8TensorAttributeJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, Float8Tensor):
+            tensor_attr_dict = {}
+
+            for tensor_attribute_name in o.optional_tensor_attribute_names:
```

Comment:
to generalize, both `tensor_attribute_names` and `optional_tensor_attribute_names` should be handled